### PR TITLE
Hide keyboard when OTP entries are blocked

### PR DESCRIPTION
**Pull request by Honey14**
https://app.shortcut.com/simpledotorg/story/5038/hide-keyboard-when-otp-entries-are-blocked

### CODE REVIEW

File: app/src/test/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtectionTest.kt [hunk 1]
```diff
@@ -64,7 +64,7 @@ class BruteForceOtpEntryProtectionTest {
     bruteForceOtpEntryProtection.incrementFailedOtpAttempt()
 
     verify(preferenceState).set(BruteForceOtpProtectionState(
-        failedLoginOtpAttempt = config.limitOfFailedAttempts + 1,
+        failedLoginOtpAttempt = config.limitOfFailedAttempts,
```
Comment (vinaysshenoy) [inline app/src/test/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtectionTest.kt:67]: Hmm, how is reducing the number of limits reached sufficient to test the new behaviour? We added a new branch, but no new tests have been added, so I assume we are missing at least one more test to verify the new branch, right?

**Review by vinaysshenoy — CHANGES_REQUESTED** [review]

File: app/src/test/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtectionTest.kt (hunk 1 above)
Comment (Honey14) [reply app/src/test/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtectionTest.kt:67]: So +1 was the problem. After the user has reached the max attempts, the blocked at time wasn't getting updated but the number of attempts was getting updated. I think the only thing we need to update in this test is the test name.
  fun `when otp limit was already reached then the blocked-at time and the number of attempts should not be updated`

Comment (Honey14) [reply app/src/test/java/org/simple/clinic/enterotp/BruteForceOtpEntryProtectionTest.kt:67]: If you look at `BruteForceOtpEntryProtectionTest`, all three scenarios are covered.

**Review by vinaysshenoy — APPROVED** [review]

**Pull request merged**